graql/reasoner/state/DisjunctiveState.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (C) 2020 Grakn Labs
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+package grakn.core.graql.reasoner.state;
+
+import grakn.core.concept.answer.ConceptMap;
+import grakn.core.graql.reasoner.explanation.DisjunctiveExplanation;
+import grakn.core.graql.reasoner.query.DisjunctiveQuery;
+import grakn.core.graql.reasoner.query.ReasonerAtomicQuery;
+import grakn.core.kb.concept.api.Concept;
+import grakn.core.kb.graql.reasoner.unifier.Unifier;
+import graql.lang.statement.Variable;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Set;
+
+public class DisjunctiveState extends AnswerPropagatorState<DisjunctiveQuery> {
+
+    // TODO Surely the subGoals won't necessarily be atomic and may be whole conjunctions
+    public DisjunctiveState(DisjunctiveQuery query, ConceptMap sub, Unifier u, AnswerPropagatorState parent, Set<ReasonerAtomicQuery> subGoals) {
```

Comment:
why TODO? subgoals will be atomic unless you go with a different resolution algorithm